AI/VCAI/Goals.cpp

```diff
@@ -1001,7 +1003,11 @@ TGoalVec Conquer::getAllPossibleSubgoals()
 							if (obj->ID.num == Obj::HERO) //enemy hero may move to other position
 								ret.push_back(sptr(Goals::VisitHero(obj->id.getNum()).sethero(h).setisAbstract(true)));
 							else //just visit that tile
-								ret.push_back(sptr(Goals::VisitTile(dest).sethero(h).setisAbstract(true)));
+								if(obj->ID.num == Obj::TOWN)
+									//if target is town, fuzzy system will use additional "estimatedReward" variable to increase priority a bit
+									ret.push_back(sptr(Goals::VisitTile(dest).sethero(h).setobjid(obj->ID.num).setisAbstract(true)));

```

Comment:
Point of this line is to set objID so it gets detected and evaluated by fuzzylite. Not sure if this check is needed, as is there is copy of it in visitTile's internals.